File: leetcode/070-climbing-stairs/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int climbStairs(int n)
    {
        if (n <= 2)
            return n;
        int prev2 = 1, prev1 = 2;
        for (int i = 3; i <= n; i++)
        {
            int cur = prev1 + prev2;
            prev2 = prev1;
            prev1 = cur;
        }
        return prev1;
    }
};

int main()
{
    Solution sol;
    cout << sol.climbStairs(2) << endl;  // 2
    cout << sol.climbStairs(3) << endl;  // 3
    return 0;
}

File: leetcode/070-climbing-stairs/solution.py
# time complexity: O(n)
# space complexity: O(1)
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2:
            return n
        prev2, prev1 = 1, 2
        for _ in range(3, n + 1):
            prev2, prev1 = prev1, prev1 + prev2
        return prev1


if __name__ == "__main__":
    print(Solution().climbStairs(2))  # 2
    print(Solution().climbStairs(3))  # 3
